templates/pkg/resource/manager.go.tpl

```diff
@@ -266,6 +263,9 @@ func (rm *resourceManager) onError(
 	r *resource,
 	err error,
 ) (acktypes.AWSResource, error) {
+	if ackcompare.IsNil(r) {
+		return nil, err
+	}
```

Comment:
Shouldn't this work just with a simple nil comparison? `if r != nil { ... }`
/cc @vijtrip2 @surajkota

Reply:
Yes, @A-Hilaly is correct. We do not need the `ackcompare.IsNil` here. `r` is a `*resource`, which is a pointer type, not an interface.

Reply:
isNil method [handles this case](https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/compare/nil.go#L36-L38) as well

Reply:
@surajkota no, the `ackcompare.IsNil` should not be used here.